x-pack/plugins/log_explorer/public/components/flyout_detail/sub_components/hover_action.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiFlexGroup,
+  EuiToolTip,
+  EuiButtonIcon,
+  useEuiTheme,
+  EuiTextTruncate,
+} from '@elastic/eui';
+import type { IconType } from '@elastic/eui';
+
+export interface HoverActionType {
+  id: string;
+  tooltipContent: string;
+  iconType: IconType;
+  onClick: () => void;
+  display: boolean;
+}
+
+interface HoverActionProps {
+  displayText: string;
+  actions: HoverActionType[];
+  width: number;
+}
+
+export const HoverAction = ({ displayText, actions, width }: HoverActionProps) => {
+  const { euiTheme } = useEuiTheme();
+  return (
+    <EuiFlexGroup
+      responsive={false}
+      alignItems="center"
+      justifyContent="flexStart"
+      gutterSize="s"
+      css={{
+        ':hover, :focus-within': {
+          '.visibleOnHoverFocus': {
+            opacity: 1,
+            visibility: 'visible',
+          },
+        },
+      }}
+    >
+      <EuiTextTruncate text={displayText} truncation="end" width={width} />
```

Comment:
**nit:** doing so we'll lose the applied marks for highlight, should we use the previous approach setting the HTML content?